app/shared/utils/utils.js

```diff
@@ -39,5 +39,19 @@ angular.module('utils', [])
       return list.splice(index, 1).length > 0;
     };
 
+    /**
+    * Transform an object into url params.
+    * ONLY knows what to do with a flat params object.
+    * Similar to jQuery.param
+    * @param {Object} params - Object of params to serialize.
+    * @returns {String} query - a url querystring (is prefixed with '?')
+    */
+    Utils.param = function (params ) {
+      var query = _.any(params) ? '?' : '';
```

Comment:
This function is difficult to understand at a glance. Make the conditions more explicit and use native functions, something like:
```js
var query = '';
var keys = Object.keys(params);
if (keys.length >= 0) {
  query = keys
    .map(function (key) {
      return key + '=' + params[key];
    })
    .join('&');
}
return query;
```